client/engine/store/mockstore.go

```diff
@@ -129,7 +129,12 @@ func (ms *MockStore) SetObjective(obj protocols.Objective) error {
 			if err != nil {
 				return fmt.Errorf("error setting channel %s from objective %s: %w", ch.Id, obj.Id(), err)
 			}
-
+		case *consensus_channel.ConsensusChannel:
+			ch := rel.(*consensus_channel.ConsensusChannel)
```

Comment:
I'm not able to _suggest_ it here because of the specific change set, but I came across this the other day:

```
		switch ch := rel.(type) {
		case *channel.Channel:
			err := ms.SetChannel(ch)
			if err != nil {
				return fmt.Errorf("error setting channel %s from objective %s: %w", ch.Id, obj.Id(), err)
			}
		case *consensus_channel.ConsensusChannel:
			err := ms.SetConsensusChannel(ch)
			if err != nil {
				return fmt.Errorf("error setting consensus channel %s from objective %s: %w", ch.Id, obj.Id(), err)
			}
		default:
			return fmt.Errorf("unexpected type: %T", rel)
		}
```

this assigns the type-asserted `ch` (it has the correct type in each block) just once, and deduplicates some of the assertions.

Reply:
Updated in [81d0a07](https://github.com/statechannels/go-nitro/pull/566/commits/81d0a07ea4beb7e61e733cad48a2975c8f65b6d3)